pkg/controller/virtinformers.go

```diff
@@ -424,20 +423,13 @@ func (f *kubeInformerFactory) VirtualMachineSnapshot() cache.SharedIndexInformer
 			"vm": func(obj interface{}) ([]string, error) {
 				vms, ok := obj.(*snapshotv1.VirtualMachineSnapshot)
 				if !ok {
-
 					return nil, unexpectedObjectError
 				}
 
-				if vms.Spec.Source.APIGroup != nil {
-					gv, err := schema.ParseGroupVersion(*vms.Spec.Source.APIGroup)
-					if err != nil {
-						return nil, err
-					}
-
-					if gv.Group == kubev1.GroupName &&
-						vms.Spec.Source.Kind == "VirtualMachine" {
-						return []string{vms.Spec.Source.Name}, nil
-					}
+				if vms.Spec.Source.APIGroup != nil &&
+					*vms.Spec.Source.APIGroup == kubev1.GroupName &&
+					vms.Spec.Source.Kind == "VirtualMachine" {
+					return []string{fmt.Sprintf("%s/%s", vms.Namespace, vms.Spec.Source.Name)}, nil
```

Comment:
Not really bugfix but definite optimization:  by adding namespace we'll make sure that controller won't queue items with same name but different namespace